### PR TITLE
use odb api in more passes, upgrade odb

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package io.shiftleft.semanticcpg.passes.cfgdominator
 
-class CfgDominator[Node](adapter: CfgAdapter[Node]) {
+class CfgDominator[N](adapter: CfgAdapter[N]) {
```
Comment (ml86) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala:3]: Please leave the speaking type parameter names in place. If `Node` collides with rename it to `NodeType`.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ class CfgDominator[Node](adapter: CfgAdapter[Node]) {
     * The used algorithm is from: "A Simple, Fast Dominance Algorithm" from
     * "Keith D. Cooper, Timothy J. Harvey, and Ken Kennedy".
     */
-  def calculate(cfgEntry: Node): Map[Node, Node] = {
+  def calculate(cfgEntry: N): Map[N, N] = {
```
Comment (ml86) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala:13]: Same as above.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala [hunk 3]
```diff
@@ -87,10 +87,10 @@ class CfgDominator[Node](adapter: CfgAdapter[Node]) {
     finger1
   }
 
-  private def createPostOrderNumbering(cfgEntry: Node): Map[Node, Int] = {
+  private def createPostOrderNumbering(cfgEntry: N): Map[N, Int] = {
```
Comment (ml86) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala:90]: Same as above.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominatorFrontier.scala [hunk 4]
```diff
@@ -2,7 +2,7 @@ package io.shiftleft.semanticcpg.passes.cfgdominator
 
 import scala.collection.mutable
 
-class CfgDominatorFrontier[Node](cfgAdapter: CfgAdapter[Node], domTreeAdapter: DomTreeAdapter[Node]) {
+class CfgDominatorFrontier[N](cfgAdapter: CfgAdapter[N], domTreeAdapter: DomTreeAdapter[N]) {
```
Comment (ml86) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominatorFrontier.scala:5]: Same as above.

**Comment by mpollmeier** [comment]
renamed `N` to `NodeType`, please re-review

**Pull request merged**